llvm/docs/LangRef.rst

```diff
@@ -3143,7 +3143,10 @@ as follows:
     specified, the default index size is equal to the pointer size. All sizes
     are in bits. The address space, ``n``, is optional, and if not specified,
     denotes the default address space 0. The value of ``n`` must be
-    in the range [1,2^24).
+    in the range [1,2^24). The fifth parameter ``<sentinel>`` specifies the
+    sentinel value of the pointer for the corresponding address space. It
+    currently accepts two values: ``0`` for an all-zero value and ``f`` for a
+    full-bit set value. The default sentinel pointer value is all-zero.
```

Comment:
"no sentinel pointer" should also be a valid option. We also should be able to specify the default for unlisted address spaces (I thought the old PR had that part implemented?)

I'm also not sure if we should just allow an arbitrary bit pattern here.

This also should define what the sentinel pointer means. Also clarify how this interacts with null_pointer_is_valid and the various flavors of nonnull